src/main/java/com/github/susom/database/DatabaseImpl.java

```diff
@@ -206,14 +207,29 @@ public void dropTableQuietly(/*@Untainted*/ @Nonnull String tableName) {
 
   @Override
   public boolean tableExists(@Nonnull String tableName) throws DatabaseException {
+
+    String schemaName = null;
     try {
-      return tableExists( tableName,
-        connection.getSchema());    // use default schema
-    } catch (SQLException exc) {
-      throw new DatabaseException("Unable to look up table " + tableName +
-        " with default db connection schema & catalog: " + exc.getMessage(),
-        exc);
+      // Use reflections to see if connection.getSchema API exists. It should exist for any JDBC7 or later implementation
+      // We still support Oracle 11 with odbc6, however, so we can't assume it's there.
+      Method method = connection.getClass().getMethod("getSchema");
+      if (method != null) {
+        schemaName = (String) method.invoke(connection);
+      } else if ("oracle".equals(this.flavor().toString())) {
```

Comment:
Cleaner to say `flavor() == Flavor.oracle` since these are defined as enumerations

Reply:
This change will be in next check in